### DISCUSSION TITLE
[Feature] When searching by faces, a checkbox to include ONLY the people selected

### DISCUSSION THREAD

**Discussion by Roachforlife** [immich-app/immich#14486] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Currently if say 2 people are selected in the face search, this returns pictures with those 2 people however the feature request would be to add a checkbox to the search screen so it would return photos with ONLY those 2 people in the photo.

Example. I select a Father and Daughter in the search, I currently may get photos that have Father and Daughter, but also it may have a 3rd or 4th person in the photos. There are times when looking for just the two individuals in the photo and no others. This feature would limit the search to be more specific.

Thanks for all the hard work on this amazing tool

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
#10373

**Comment by nosajthenitram** [comment]
I've started to look at this feature. One minor limitation will be using the existing asset_face table for determining the presence of only the individuals tagged, if there are people in the photo that are facing backwards or to the side, they will not be identified as a face and therefore show up in your searches.

**Reply by nosajthenitram** [reply]
@alextran1502 Thoughts on this implementation? Strict person search toggle at the end of the people filter search bar. When it's selected it turns red. 

<img width="1015" height="62" alt="image" src="https://github.com/user-attachments/assets/f240e699-2f7b-4a42-9b7a-c50cc1f0ddbc" />

<img width="1018" height="72" alt="image" src="https://github.com/user-attachments/assets/0450a905-bc5a-44fa-95db-6f542965e9ab" />

or another option is to use the checkbox for consistency in the modal

<img width="998" height="87" alt="image" src="https://github.com/user-attachments/assets/06a3d242-73f1-4c94-bcec-daf159fcd4bb" />


Search will return only assets with the exact people selected

**Reply by Roachforlife** [reply]
I dig it

**Comment by ychongfoo** [comment]
This request aligns with a feature I use often in both Sxxxxng Gallery and Gxxxle Photos (whose name we should not mention), where searching by faces can be refined with an “Only Them” toggle. That option shows only the exact people selected (excluding others).

I also came across [#10373](https://github.com/immich-app/immich/discussions/10373), which touches on related functionality.

I’ve opened a refinement request here: #20949  — proposing an explicit “Only Them” toggle (Sxxxxng/Gxxxle Photos–style) for clearer UX and discoverability. I think this would complement your idea nicely and make it more intuitive for everyday users.

**Comment by grillop** [comment]
I'll take any solution. Can live with photos of only the 2 but other undetected etc.  Might even prefer it...
